packages/langium-cli/src/generate.ts

```diff
@@ -58,45 +58,74 @@ export function generate(config: LangiumConfig): boolean {
 
     const output = path.join(relPath, config.out ?? 'src/generated');
     console.log(`${getTime()}Writing generated files to ${output.white.bold}`);
-    mkdirWithFail(output);
+
+    if (await rmdirWithFail(output, ['ast.ts', 'grammar.ts', 'grammar-access.ts', 'parser.ts', 'module.ts'])) {
+        return false;
+    }
+    if (await mkdirWithFail(output)) {
+        return false;
```

Comment:
Wouldn't it be easier to reject the promise?

Reply:
Well, kind of. We would have to deal with the `generate` function throwing inside of the other code, which would be quite annoying (especially in the watch mode). So I went with this direction.

Reply:
I see. I think the code would be better understandable if you use a more specific return type:
```typescript
export type ResultType = 'success' | 'failure';
```
or something like that.